low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +45,129 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        self.server_map = {
+            "memory": self.get_mem_info
+            }
+
+    @staticmethod
+    def get_node_details(node):
+        """
+        Parse node information and returns left string and instance.
+        Example
+            "compute"    -> ("compute", "*")
+            "compute[0]" -> ("compute", "0")
+        """
+        res = re.search(r"(\w+)\[([\d]+)\]|(\w+)", node)
+        inst = res.groups()[1] if res.groups()[1] else "*"
+        node = res.groups()[0] if res.groups()[1] else res.groups()[2]
+        return node, inst
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            info["last_updated"] = int(time.time())
+            for fru in self.server_map:
+                info.update({fru: self.server_map[fru]()})
+        else:
+            fru = None
+            fru_found = False
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_map.get(fru):
+                    fru_found = True
+                    info = self.server_map[fru]()
+                    break
+        if not fru_found:
+            raise HealthProviderError(
```

Comment:
Rebase your PR.
HealthProviderError is renamed as ResourceMapError in error.py

Reply:
Done